lib/main.js

```diff
@@ -38,61 +40,68 @@ module.exports = class Client {
         return new Promise((resolve, reject) => {
             const writer = new Writer(files, options, false, true);
 
-            this.transforms.forEach((entry) => {
+            this.transforms.forEach(entry => {
                 writer.transform(entry.transform, entry.options);
             });
 
-            this.plugins.forEach((entry) => {
+            this.plugins.forEach(entry => {
                 writer.plugin(entry.plugin, entry.options);
             });
 
             writer
                 .bundle()
                 .pipe(JSONStream.stringify())
-                .pipe(request.post({
-                    url: url.resolve(this.options.buildServerUri, 'feed'),
+                .pipe(
+                    request.post(
+                        {
+                            url: url.resolve(this.options.buildServerUri, 'feed'),
+                            json: true,
+                        },
+                        (error, response, body) => {
+                            if (error) {
+                                return reject(error);
+                            }
+                            if (response.statusCode === 200) {
+                                return resolve(body);
+                            }
+                            if (response.statusCode === 400) {
+                                return reject(new Error(body.message));
+                            }
+                            reject(new Error(`Asset build server responded with unknown error. Http status ${response.statusCode}`));
+                        }
+                    )
+                );
+        });
+    }
+
+    /**
+     * Make a bundle out of asset feeds on the asset server
+     */
+
+    createRemoteBundle (sources) {
+        return new Promise((resolve, reject) => {
+            request.post(
+                {
+                    url: url.resolve(this.options.buildServerUri, 'bundle'),
+                    body: sources,
```

Comment:
Missing timeouts

Reply:
You're on an old commit.
See: https://github.com/asset-pipe/asset-pipe-client/pull/29/files#diff-16b7d75b70953cbf4755508170f2f24cR25
However, not sure if just chucking in `timeout: 5000` (current solution) is enough? Do we need retries? If so, howtoretrywhenstreamsareinvolved?